### DISCUSSION TITLE
Using new Caddy proxy with Traefik

### DISCUSSION THREAD

**Discussion by ericmoon** [inventree/InvenTree#6756] [Q&A] [answered]
Has  anyone got the Caddy proxy working with Traefik?  If so, can you provide your caddyfile and docker-compose.yml?

I had everything working fine on the <= 13.x versions, but I had to reinstall the server, and now I'm stuck in a redirect failure.

**Comment by SchrodingersGat** [comment]
@ericmoon if the previous setup worked with the nginx setup file, no reason to stop using that - it will still work just fine.

If you are running the Caddy setup behind another proxy, you will need to ensure that the Caddyfile is adjusted to listen to all relevant hosts

**Comment by ericmoon** [comment]
Well, after beating my head against the monitor for a couple of days on and off, I'm going to punt on this one.  I have it working on my internal network, but adding Traefik in front of it doesn't let me access externally, no matter the config I've tried.

I can bypass caddy and access it externally, but of course, I lose the static and media serves, so it's not usable.

**Reply by SchrodingersGat** [reply]
@ericmoon I have no experience with Traefik, so cannot really comment on that, but if your previous nginx config was working well, I'd suggest continuing to use that!

You might also want to look at [this similar issue](https://github.com/inventree/InvenTree/issues/6720) for inspiration...

**Comment by ericmoon** [comment]
Coming back to this after a long break -  I have it working now, with a slight wrinkle:
If I try to login to the 'standard' UI, I immediately get logged out, but...
If I try to login via the new Platform UI, it works like a charm, no issues.

So, two questions -
  Is there a significant difference in handling authentication between the two UIs?
  Since the plan is to switch to the Platform UI, should I bother with any debug logout issue?

**Reply by wolflu05** [reply]
There were lots of changes recently with how authentication is implemented in PUI. You would need to provide version information of your instance so we can tell you if PUI will still function for the next time. If you're still running PUI with auth tokens (before #6970), there is a high chance, that when updating, auth will work like you're experiencing it with CUI.

**Reply by ericmoon** [reply]
Fresh docker install pulled today (no previous install/data).
Version: 0.16.0 dev
API:202
Commit: 80c3174

**Reply by wolflu05** [reply, marked as answer]
ok, that already uses the (new) session auth in PUI. This issue seams similar: #7387 , so I would recommend you following on that and see if you have the same issue.

**Reply by ericmoon** [reply]
@wolflu05 That fixed the last issue I was - Thanks!
